pkg/workflows/utils.go

```diff
@@ -84,3 +86,14 @@ func HashTruncateName(name string) [10]byte {
 
 	return result
 }
+
+// HexDecodeWorkflowName makes the best effort to decode the hex workflow name.
+// In case of failure it will log the error and return the input name.
+func HexDecodeWorkflowName(encoded string, logger logger.Logger) string {
+	decoded, err := hex.DecodeString(encoded)
+	if err != nil {
+		logger.Errorf("failed to decode WorkflowName %q: %v", encoded, err)
+		return encoded
```

Comment:
Passing a logger as a function argument is usually a code smell. Where is this used?